EventKit/eventkit-cloud
# Filter by favorites and refactor filter window UI

# Description of Improvement

Adds ability to filter provider list by favorites, and cleans up UI to match updated mockups for filter window.

## How to test

Verify favorites can be set while selected + unselected in product list.
Verify filter by favorites honors favorite state of product.
Verify sort by favorites honors favorite state of product.
Verify clear all resets filter by favorites.
Verify favorite state of product is honored on refresh + change step of workflow.

## Quality Assurance

The following items have been updated:


## Screenshots
<img width="1786" alt="Screen Shot 2022-09-06 at 11 31 47 AM" src="https://user-images.githubusercontent.com/3164631/188675970-5a7e0368-ab54-4173-b3c0-d0ea48e02fb8.png">

:smile:
